JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeScreens.kt

```diff
@@ -60,76 +66,176 @@ import com.example.jetnews.R
+    onRefreshPosts: () -> Unit,
+    onErrorDismiss: (Long) -> Unit,
+    onInteractWithList: () -> Unit,
+    onInteractWithDetail: (String) -> Unit,
     openDrawer: () -> Unit,
-    scaffoldState: ScaffoldState = rememberScaffoldState()
+    homeListLazyListState: LazyListState,
+    articleDetailLazyListStates: Map<String, LazyListState>,
+    scaffoldState: ScaffoldState
 ) {
-    // UiState of the HomeScreen
-    val uiState by homeViewModel.uiState.collectAsState()
+    HomeScreenWithList(
+        uiState = uiState,
+        onRefreshPosts = onRefreshPosts,
+        onErrorDismiss = onErrorDismiss,
+        openDrawer = openDrawer,
+        homeListLazyListState = homeListLazyListState,
+        scaffoldState = scaffoldState
+    ) { hasPostsUiState, modifier ->
+        Row {
+            PostList(
+                postsFeed = hasPostsUiState.postsFeed,
+                onArticleTapped = onSelectPost,
+                favorites = hasPostsUiState.favorites,
+                onToggleFavorite = onToggleFavorite,
+                contentPadding = rememberInsetsPaddingValues(
+                    insets = LocalWindowInsets.current.systemBars,
+                    applyTop = false,
+                    applyEnd = false,
+                ),
+                modifier = modifier
+                    .width(334.dp)
+                    .pointerInput(Unit) {
+                        while (currentCoroutineContext().isActive) {
+                            awaitPointerEventScope {
+                                awaitPointerEvent(PointerEventPass.Initial)
+                                onInteractWithList()
+                            }
+                        }
+                    },
+                state = homeListLazyListState
+            )
+            // Crossfade between different detail posts
+            Crossfade(targetState = hasPostsUiState.selectedPost) { detailPost ->
+                // Get the lazy list state for this detail view
+                val detailLazyListState by derivedStateOf {
+                    articleDetailLazyListStates.getValue(detailPost.id)
+                }
+
+                // Key against the post id to avoid sharing any state between different posts
+                key(detailPost.id) {
+                    PostContent(
+                        post = detailPost,
+                        modifier = modifier
+                            .fillMaxSize()
+                            .pointerInput(Unit) {
+                                while (currentCoroutineContext().isActive) {
+                                    awaitPointerEventScope {
+                                        awaitPointerEvent(PointerEventPass.Initial)
+                                        onInteractWithDetail(detailPost.id)
+                                    }
+                                }
+                            },
+                        contentPadding = rememberInsetsPaddingValues(
```

Comment:
This content padding is the same as above, should we reuse it? we can extract it out to a `contentPadding` variable

Reply:
These are subtly different, due to the `applyStart` vs `applyEnd`. Although I need to update this logic anyway now that the nav rail is involved as well, so thanks for the call out!

Reply:
Updated with the latest commits.